### DISCUSSION TITLE
Git config `submodule.recurse=true` has unintended side effects

### DISCUSSION THREAD

**Discussion by mtardy** [renovatebot/renovate#26153] [Request Help] [answered]
### How are you running Renovate?

None

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

None

### Describe the problem

This is a follow-up of discussion https://github.com/renovatebot/renovate/discussions/25689, transformed into issue https://github.com/renovatebot/renovate/issues/26057 that ended up with PR https://github.com/renovatebot/renovate/pull/26076. It is a discussion created after comment https://github.com/renovatebot/renovate/issues/26057#issuecomment-1840453677.

This fix fixed my issue and now I can update my stable branch which is amazing, thank you for that. However I discovered a side effect of `submodule.recurse=true`: as you see [in the documentation](https://git-scm.com/docs/git-config#Documentation/git-config.txt-submodulerecurse), it affects a lot of other commands (than `checkout`) that can use flags that are incompatible with `--recurse-submodules`.

I discovered that one of my code generation script was using `git grep --untrack <something>` and that generates:
```
fatal: --untracked not supported with --recurse-submodules
```

A workaround can be adding `--no-recurse-submodules` to specific command or use `git -c submodule.recurse=0` for commands that don't support `--no-recurse-submodules` but the issue is that I vendor those scripts and it's not really convenient to modify them.

The fix would be to use `--recurse-submodule` only when needed (when changing to another branch, using `checkout` or `switch`, I don't know renovate's internals, here is [a list of all the affected commands](https://git-scm.com/docs/git-config#Documentation/git-config.txt-submodulerecurse)) and not set it as a global config, my previous suggestion was not anticipating those issues.

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment, marked as answer]
Revert PR: #26157
